src/include/schema.h

```diff
@@ -228,15 +229,21 @@ struct __wt_import_list {
                 WT_SESSION_LOCKED_TABLE));                                                    \
         WT_WITH_LOCK_WAIT(session, &S2C(session)->schema_lock, WT_SESSION_LOCKED_SCHEMA, op); \
     } while (0)
-#define WT_WITH_SCHEMA_LOCK_NOWAIT(session, ret, op)                               \
-    do {                                                                           \
-        WT_ASSERT(session,                                                         \
-          FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_SCHEMA) ||              \
-            !FLD_ISSET(session->lock_flags,                                        \
-              WT_SESSION_LOCKED_HANDLE_LIST | WT_SESSION_NO_SCHEMA_LOCK |          \
-                WT_SESSION_LOCKED_TABLE));                                         \
-        WT_WITH_LOCK_NOWAIT(                                                       \
-          session, ret, &S2C(session)->schema_lock, WT_SESSION_LOCKED_SCHEMA, op); \
+#define WT_WITH_SCHEMA_LOCK_NOWAIT(session, ret, op)                                           \
+    do {                                                                                       \
+        WT_ASSERT(session,                                                                     \
+          FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_SCHEMA) ||                          \
+            !FLD_ISSET(session->lock_flags,                                                    \
+              WT_SESSION_LOCKED_HANDLE_LIST | WT_SESSION_NO_SCHEMA_LOCK |                      \
+                WT_SESSION_LOCKED_TABLE));                                                     \
+        int lock_ret;                                                                          \
+        WT_WITH_LOCK_NOWAIT(                                                                   \
+          session, ret, lock_ret, &S2C(session)->schema_lock, WT_SESSION_LOCKED_SCHEMA, op);   \
+        if (lock_ret == EBUSY) {                                                               \
+            WT_IGNORE_RET(__wt_session_set_last_error(session, EBUSY, WT_CONFLICT_SCHEMA_LOCK, \
+              "another thread is currently accessing the schema"));                            \
+            ret = EBUSY;                                                                       \
+        }                                                                                      \
```

Comment:
Why don't we make this general?
```suggestion
        if (lock_ret != 0) {                                                               \
            WT_IGNORE_RET(__wt_session_set_last_error(session, lock_ret, WT_CONFLICT_SCHEMA_LOCK, \
              lock_ret == EBUSY ? "another thread is currently accessing the schema" : "TBD));                            \
            ret = lock_ret;                                                                       \
        }                                                                                      \
```

Reply:
Done in [Refactored tests and added suggested changes from CR](https://github.com/wiredtiger/wiredtiger/pull/11490/commits/321cd5a25d6d0807e899dbf1cdcc9d20a29ff344).

Reply:
@jiechenbo do you think the sub level error code `WT_CONFLICT_SCHEMA_LOCK` is still appropriate in this case? Or should it be limited to `EBUSY` errors only, and have `WT_NONE` otherwise?

Reply:
Conclusion after discussing in-person - `WT_CONFLICT_SCHEMA_LOCK` is not appropriate, since if the error code is not `EBUSY` it means that the problem does not originate from the lock being held by another thread (i.e. no conflict). In the case where `__lock_ret != EBUSY`, the sub level error code should be `WT_NONE`.

Reply:
Done in [Added some comments and altered the lock macros to handle non-EBUSY r…](https://github.com/wiredtiger/wiredtiger/pull/11490/commits/e99c793a52760a3411832ef47e30d2d2d3578bd4).